js/components/buttonMaker.js

```diff
@@ -1,40 +1,43 @@
 /**
  * Creates a button of a specific type.
- * @param {string} type - The type of button ("begin", "choose", "leftArrow", "rightArrow").
+ * @param {object} props - The properties of the button.
+ * @param {string} props.type - The type of button ("proceed", "back", "leftArrow" or "rightArrow").
+ * @param {string} props.text - The text to be displayed on the button.
+ * @param {void} props.onClick - Function to be executed on clicking the button.
  * @returns {HTMLElement} - The generated button element.
- * @throws {Error} - Throws an error if the type is not "begin", "choose", "leftArrow" or "rightArrow".
+ * @throws {Error} - Throws an error if the type is not "proceed", "back" or "leftArrow" or "rightArrow".
  */
 
-export function buttonMaker(type) {
-   if (type !== 'begin' && type !== 'choose' && type !== 'leftArrow' && type !== 'rightArrow') {
-      throw new Error(
-         'Invalid button type. Only "begin", "choose", "leftArrow" or "rightArrow" are allowed.',
-      );
+export function buttonMaker(props) {
+   const type = props.type;
+   const text = props.text;
+
+   if (type !== 'proceed' && type !== 'back' && type !== 'rightArrow' && type !== 'leftArrow') {
+      throw new Error('Invalid button type. Only "proceed", "back" or "arrow" are allowed.');
    }
 
    let button = document.createElement('button');
-   let buttonImg = document.createElement('img');
-   button.appendChild(buttonImg);
-
-   switch (type) {
-      case 'begin':
-         buttonImg.src = 'assets/buttonBegin.svg';
-         break;
-      case 'choose':
-         buttonImg.src = 'assets/buttonChoose.svg';
-         break;
-      case 'leftArrow':
-         buttonImg.src = 'assets/buttonNext.svg';
-         button.style.transform = 'rotate(180deg)';
-         break;
-      case 'rightArrow':
-         buttonImg.src = 'assets/buttonNext.svg';
-         break;
-      default:
-         break;
+   button.classList.add('bigText');
```

Comment:
**nitpick:** You started to have quite a big `ifology` here. ArrowButton should be a different component, than this normal button with text. You will avoid doing this ifs, and your code will be much more clear, and understandable. Create `buttonIconMaker`, or `buttonArrowMaker`, so it makes only this small button with arrow